types/compact_vector_description.h

```diff
@@ -28,64 +28,67 @@ struct CompactVectorDescription
     {
     }
 
-    virtual void parseJson(void * val, JsonParsingContext & context) const
+    void parseJson(void * val, JsonParsingContext & context) const override
```

Comment:
Sure, it's redundant, just like comments and parameter names.  The first is redundant in the parameter names, but which is better?

```
void transferMoney(int,int,int)
void transferMoney(int fromAccountNumber, int toAccountNumber, int amountInCents)
```

For virtual, it is important for a programmer to know which methods are virtual, as any of the following actions could have unexpected and difficult to debug side-effects:
- renaming a method
- changing a signature
- deleting a method
- cleaning up redundant versions of a method

Without the redundant virtual keyword, a programmer needs to find the original .h file and look to see if a method is virtual before modifying it.  Of course they won't, and we will end up with bugs.  With the virtual keyword there, it is clearly documented as being virtual and so extra care needs to be taken.